lib/api/src/modules/refs.ts

```diff
@@ -113,46 +123,73 @@ export const init: ModuleFn = ({ store, provider, fullAPI }) => {
     checkRef: async (ref) => {
       const { id, url } = ref;
 
-      const handler = async (response: Response) => {
-        if (response) {
-          const { ok } = response;
-
-          if (ok) {
-            return response.json().catch((error: Error) => ({ error }));
-          }
-        } else {
-          logger.warn('an auto-injected ref threw a cors-error');
-        }
+      const loadedData: { error?: Error; stories?: StoriesRaw } = {};
 
-        return false;
-      };
-
-      const [stories, metadata] = await Promise.all([
+      const [included, omitted, iframe] = await allSettled([
         fetch(`${url}/stories.json`, {
           headers: {
             Accept: 'application/json',
           },
           credentials: 'include',
-        })
-          .catch(() => false)
-          .then(handler),
-        fetch(`${url}/metadata.json`, {
+        }),
+        fetch(`${url}/stories.json`, {
           headers: {
             Accept: 'application/json',
           },
-          credentials: 'include',
-          cache: 'no-cache',
-        })
-          .catch(() => false)
-          .then(handler),
+          credentials: 'omit',
+        }),
+        fetch(`${url}/iframe.html`, {
+          credentials: 'omit',
+        }),
```

Comment:
I'm not sure about this check. It won't work in the follow situations:

1. The server doesn't allow cross-origin fetch access to the `iframe.html` at all (this is the case for Chromatic, although in Chromatic's case `stories.json` is fetchable as `included`).

2. The server isn't setup for CORS at all (i.e. serves no `Access-Control-Allow-Origin` header). This is sort of the same as 1, functionally.

3. The server requires login to get `iframe.html`. We could probably assume such servers will serve `stories.json` however, maybe.

Given we aren't looking at the result of the `fetch`, perhaps we should use `mode: 'no-cors'`  for this request? It won't fix 3. above but maybe we are OK with that for the reason mentioned?

Reply:
> The server doesn't allow cross-origin fetch access to the iframe.html at all (this is the case for Chromatic, although in Chromatic's case stories.json is fetchable as included).

So it will work then?

> The server isn't setup for CORS at all (i.e. serves no Access-Control-Allow-Origin header). This is sort of the same as 1, functionally.

correct it won't work, that's the point.

> The server requires login to get iframe.html. We could probably assume such servers will serve stories.json however, maybe.

Yes, like chromatic, or storybook has no way to load it, so will show an error. It's the best we can do?

Reply:
I'm not quite sure I am understanding, let me make a chart:

|Host type | Example | Resource public/private? | `included` | `omitted` | `iframe`|error?|
|-----|-----|-----|-----|-----|-----|-----|
|Authed, proper CORS | Chromatic | private, correct credentials | `true` | `false` | `false`|`false`|
|Authed, proper CORS | Chromatic | private, incorrect credentials | `false` | `false` | `false`|`true`|
|Authed, proper CORS | Chromatic | public | `true` | `true` | `false`|`false`|`false`|
|Simple CORS | Netlify | - | `false` | `true` | `true`|`false`|`false`|
|No CORS | S3 | - | `false` | `false` | `false`|`true`|

Doesn't this mean the S3 case is going to show an error in SB?

Reply:
s3 no cors iframe will be true now with my latest changes